packages/block-library/src/template-part/edit/selection/template-part-previews.js

```diff
@@ -111,23 +111,37 @@ function TemplatePartsByArea( {
 	area = 'uncategorized',
 	labelsByArea,
 } ) {
+	const templatePartsToShow =
+		templateParts.filter(
+			( templatePart ) =>
+				'uncategorized' === area || templatePart.area === area
+		) || [];
 	const templatePartsByArea = useMemo( () => {
-		return Object.values( groupBy( templateParts, 'area' ) );
+		return Object.values( groupBy( templatePartsToShow, 'area' ) );
 	}, [ templateParts, area ] );
```

Comment:
How about ditching the `return` here?

Reply:
Good point.  Although I just moved some other things into this memo so we will return an object now. 😁